File: a040(armstrong_numbers)/C++/armstrong_numbers.cpp
#include <iostream>
#include <cmath>

int main(void) {

  int n, m;
  while (std::cin >> n >> m) {
      if(n < 0 || n > m || m > 1000000) {
        break;
      }
      int count = 0;
      for(int number = n; number < m; number++) {
        int power = (std::to_string(number)).size();
        int total = 0;
        int temp = number; // avoid from changing number
        while(temp != 0) {
          total += pow((temp % 10), power);
          temp /= 10;
        }
        if(total == number) {
          count++;
          std::cout << number << ' ';
        }
      }
      if(count == 0) {
        std::cout << "none";
      }
      std::cout << '\n';
  }


  return 0;
}

File: a040(armstrong_numbers)/Python/armstrong_numbers.py
while True:
    try:
        n, m = map(int, input().split())
        if n == 0 or n > m or m > 1000000:
            break
        count = False
        for number in range(n, m):
            power = len(str(number))
            if number == sum([int(n) ** power for n in str(number)]):
                count = True
                print(number, end = ' ')
        if not count:
            print('none', end = '')
        print()
    except:
        break
